target/arm/cpu.c

```diff
@@ -136,6 +136,27 @@ static void cp_reg_reset(gpointer key, gpointer value, gpointer opaque)
     ARMCPRegInfo *ri = value;
     ARMCPU *cpu = opaque;
 
+#ifdef TARGET_CHERI
+    if (cpreg_field_is_cap(ri)) {
+        assert(ri->fieldoffset && "cpreg cap without offset?");
+        /*
+         * We ensure that all cap_register_t values are reset to a canonical
+         * capability rather than all zeroes. This ensures that values such
+         * as pesbt and capreg_state are initialized correctly.
+         * The default reset value is null unless has_special_capresetvalue is
+         * set.
+         * TODO: We do this even for (ARM_CP_SPECIAL | ARM_CP_ALIAS) since
```

Comment:
Comment says TODO, but it is done. I wonder if this causes any problems? I don't think it does.

Reply:
I am a little concerned about ARM_CP_ALIAS not being ignored here. My worry is that an aliasing register will use an offset into another register, and resetting both in the wrong order could do something funny. I had a scan through and it doesn't look like there are any cap aliases that could cause this problem.